### ISSUE TITLE
Running npm test throws an exception

### ISSUE THREAD

**Issue by astorije** [w3c/echidna#34]
npm even calls this a `weird error` :)

Here the stack trace:

```
$ npm test

> echidna@0.1.0 test [...]
> mocha

[...]/test/lib/testserver.js:47
  return "http://localhost:" + server.address().port;
                                               ^
TypeError: Cannot read property 'port' of null
    at Function.TestServer.location ([...]/test/lib/testserver.js:47:48)
    at Suite.<anonymous> ([...]/test/test.js:22:51)
    at context.describe.context.context ([...]/node_modules/mocha/lib/interfaces/bdd.js:74:10)
    at Suite.<anonymous> ([...]/test/test.js:17:3)
    at context.describe.context.context ([...]/node_modules/mocha/lib/interfaces/bdd.js:74:10)
    at Object.<anonymous> ([...]/test/test.js:15:1)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at [...]/node_modules/mocha/lib/mocha.js:185:27
    at Array.forEach (native)
    at Mocha.loadFiles ([...]/node_modules/mocha/lib/mocha.js:182:14)
    at Mocha.run ([...]/node_modules/mocha/lib/mocha.js:394:31)
    at Object.<anonymous> ([...]/node_modules/mocha/bin/_mocha:394:16)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
npm ERR! weird error 8
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! not ok code 0
```

it seems that when you call the `location()` method ([here](https://github.com/w3c/echidna/blob/4a0af2b8476f16cfb35b962303de72508d126c1f/test/test.js#L30) for example) from the `TestServer`, `server.adress()` ([here](https://github.com/w3c/echidna/blob/4a0af2b8476f16cfb35b962303de72508d126c1f/test/lib/testserver.js#L48)) is not ready yet because the `app.listen(port)` call ([here](https://github.com/w3c/echidna/blob/4a0af2b8476f16cfb35b962303de72508d126c1f/test/lib/testserver.js#L44)) can be asynchronous in some cases (that's a weird thing to say "potentially asynchronous", that's where we need promises!).

[Here is an explication on Stack Overflow](http://stackoverflow.com/a/10069091/1935861). Why this passes Travis CI bugs me though.

**Comment by astorije** [comment]
It appears that I had another `node test/lib/testserver.js` running in a different terminal tab, sorry...

However, I don't think this should be a `wontfix` tag, as my use case is legit, I think: the test server running in the background was there to handle documents when I was manually testing on my local environment (i.e. running jobs on the web interface), and `npm test` should still be able to be run successfully.

**Comment by tripu** [comment]
On master, without local changes, I'm getting this right now (it's a bit different, @astorije):

```
$ mocha

echidna/test/lib/testserver.js:48
  return "http://localhost:" + server.address().port;
                                               ^
TypeError: Cannot read property 'port' of null
    at Function.TestServer.location (echidna/test/lib/testserver.js:48:48)
    at Object.<anonymous> (echidna/test/test.js:15:32)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at /usr/local/lib/node_modules/mocha/lib/mocha.js:183:27
    at Array.forEach (native)
    at Mocha.loadFiles (/usr/local/lib/node_modules/mocha/lib/mocha.js:180:14)
    at Mocha.run (/usr/local/lib/node_modules/mocha/lib/mocha.js:382:31)
    at Object.<anonymous> (/usr/local/lib/node_modules/mocha/bin/_mocha:381:16)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
```

??

**Comment by astorije** [comment]
Do you have a test server running while executing `mocha`?
Does the issue persist if you run `npm test` rather than `mocha`? Well actually, you probably need to uninstall your global mocha if it doesn't appear in your `node_modules` folder, worth checking.

**Comment by tripu** [comment]
This error seems to be gone now for me; tests run OK typing either `mocha` or `npm test`.
I don't understand what happened before&hellip;

**Comment by plehegar** [comment]
I'll look into it and fix it. Either I'll fail gracefully or change the port if I can't use the default one.

**Comment by tripu** [comment]
I guess this one's closed, now that PR #42's merged to `master`&hellip;?

**Comment by astorije** [comment]
I think so.
@plehegar, when you want a commit to automatically close a ticket, start your commit message with "Fix #{issue_number}: ...". The issue will be closed when the PR is merged with master, IIRC.
